components/client/src/test/unit/java/com/hotels/styx/client/StyxHttpClientTest.java

```diff
@@ -180,470 +107,360 @@ private static BackendService backendWithOrigins(int originPort) {
+        HttpResponse response = styxHttpClient.sendRequest(SOME_REQ).toBlocking().first();
 
-        subscription.unsubscribe();
+        ArgumentCaptor<RetryPolicy.Context> retryContext = ArgumentCaptor.forClass(RetryPolicy.Context.class);
+        ArgumentCaptor<LoadBalancingStrategy> lbStrategy= ArgumentCaptor.forClass(LoadBalancingStrategy.class);
+        ArgumentCaptor<LoadBalancingStrategy.Context> lbContext = ArgumentCaptor.forClass(LoadBalancingStrategy.Context.class);
 
-        assertThat(metricRegistry.counter("origins.App-X.requests.cancelled").getCount(), is(1L));
-        assertThat(metricRegistry.counter("origins.App-X.Origin-Y.requests.cancelled").getCount(), is(1L));
-    }
+        verify(retryPolicy).evaluate(retryContext.capture(), lbStrategy.capture(), lbContext.capture());
 
-    @Test
-    public void doesNotRetryAnAlreadyCancelledRequest() {
-        RetryPolicy retryPolicy = mock(RetryPolicy.class);
-        PublishSubject<HttpResponse> responseSubject = PublishSubject.create();
-        StyxHttpClient styxHttpClient = styxClient(nettyConnection(mock(Channel.class)), responseSubject, retryPolicy);
+        assertThat(retryContext.getValue().appId(), is(backendService.id()));
+        assertThat(retryContext.getValue().currentRetryCount(), is(1));
+        assertThat(retryContext.getValue().lastException().get(), Matchers.instanceOf(OriginUnreachableException.class));
 
-        LoggingTestSupport support = new LoggingTestSupport(StyxHttpClient.class);
-        try {
-            Subscription subscription = styxHttpClient
-                    .sendRequest(SOME_REQ)
-                    .subscribe(new TestSubscriber<>());
+        assertThat(lbStrategy.getValue(), notNullValue());
 
-            responseSubject.onNext(response(OK).build());
-            subscription.unsubscribe();
+        assertThat(lbContext.getValue().appId(), is(backendService.id()));
+        assertThat(lbContext.getValue().oneMinuteRateForStatusCode5xx(SOME_ORIGIN), is(0.0));
+        assertThat(lbContext.getValue().currentRequest(), is(SOME_REQ));
 
-            responseSubject.onError(new OriginUnreachableException(newOriginBuilder("hotels.com", 8080).build(), new RuntimeException("test")));
+        assertThat(response.status(), is(OK));
 
-            verifyZeroInteractions(retryPolicy);
-        } finally {
-            support.stop();
-        }
+        InOrder ordered = inOrder(firstClient, secondClient);
+        ordered.verify(firstClient).sendRequest(eq(SOME_REQ));
+        ordered.verify(secondClient).sendRequest(eq(SOME_REQ));
     }
 
     @Test
-    public void releasesConnectionWhenOperationExecutionFails() {
-        requestOperationFactory = newFailingRequestOperationFactory(new RuntimeException());
-
-        Connection connection = openConnection(SOME_ORIGIN);
-        ConnectionPool pool = mockPool(connection);
+    public void stopsRetriesWhenRetryPolicyTellsToStop() {
+        ConnectionPool firstPool = mockPool(ORIGIN_1);
```

Comment:
There are some very large test methods here. One way we should shrink them is by moving the `mockPool` variable declaration & assignment to the class's setup method.

Reply:
Mock pools can also be inlined.

Reply:
Tidied up this a bit.